src/schema/schema_triggers.py

```diff
@@ -1907,7 +1907,8 @@ def get_upload_datasets(property_key, normalized_type, user_token, existing_data
 def set_activity_creation_action(property_key, normalized_type, user_token, existing_data_dict, new_data_dict):
     if 'normalized_entity_type' not in new_data_dict:
         raise KeyError("Missing 'normalized_entity_type' key in 'existing_data_dict' during calling 'set_activity_creation_action()' trigger method.")
-    
+    if new_data_dict and new_data_dict['creation_action']:
```

Comment:
This condition will cause `KeyError: 'creation_action'` for when `creation_action` field is not provided in the request. Need to check the existence of the key.